src/esp/scene/HM3DSemanticScene.cpp

```diff
@@ -134,9 +134,9 @@ bool SemanticScene::buildHM3DHouse(std::ifstream& ifs,
                               uint8_t(colorInt & 0xff)};
     // object category will possibly have commas
     const std::string objCategoryName = tokens[1];
-    // room/region is always last token - get rid of first comma
-    int regionID =
-        std::stoi(Cr::Utility::String::trim(tokens[tokens.size() - 1], " ,"));
+
+    // region id comes after instace category raw text
+    int regionID = std::stoi(Cr::Utility::String::trim(tokens[2], " ,"));
```

Comment:
One thing to note here - the split that creates the `tokens` array is splitting on empty quotes.  If a text string region name, encapsulated within quotes, is added, that will also be an element in this tokens array at idx 3.